code/modules/mob/living/carbon/xenomorph/update_icons.dm

```diff
@@ -127,6 +130,282 @@
+
+/mob/living/carbon/Xenomorph/Hivelord/update_wounds()
+	remove_overlay(X_WOUND_LAYER)
+	if(health < maxHealth * 0.5) //Injuries appear at less than 50% health
+		var/image/I
+		var/is_dead = FALSE
+		if(stat == DEAD)
+			is_dead = TRUE
+		if(resting)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_resting", "layer"=-X_WOUND_LAYER)
+		else if(sleeping || is_dead)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_sleeping", "layer"=-X_WOUND_LAYER)
+		else
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded", "layer"=-X_WOUND_LAYER)
+
+		overlays_standing[X_WOUND_LAYER] = I
+		apply_overlay(X_WOUND_LAYER)
+
+
+/mob/living/carbon/Xenomorph/Praetorian/update_wounds()
+	remove_overlay(X_WOUND_LAYER)
+	if(health < maxHealth * 0.5) //Injuries appear at less than 50% health
+		var/image/I
+		var/is_dead = FALSE
+		if(stat == DEAD)
+			is_dead = TRUE
+		if(resting)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_resting", "layer"=-X_WOUND_LAYER)
+		else if(sleeping || is_dead)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_sleeping", "layer"=-X_WOUND_LAYER)
+		else
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded", "layer"=-X_WOUND_LAYER)
+
+		overlays_standing[X_WOUND_LAYER] = I
+		apply_overlay(X_WOUND_LAYER)
+
+
+/mob/living/carbon/Xenomorph/Carrier/update_wounds()
+	remove_overlay(X_WOUND_LAYER)
+	if(health < maxHealth * 0.5) //Injuries appear at less than 50% health
+		var/image/I
+		var/is_dead = FALSE
+		if(stat == DEAD)
+			is_dead = TRUE
+		if(resting)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_resting", "layer"=-X_WOUND_LAYER)
+		else if(sleeping || is_dead)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_sleeping", "layer"=-X_WOUND_LAYER)
+		else
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded", "layer"=-X_WOUND_LAYER)
+
+		overlays_standing[X_WOUND_LAYER] = I
+		apply_overlay(X_WOUND_LAYER)
+
+
+/mob/living/carbon/Xenomorph/Boiler/update_wounds()
+	remove_overlay(X_WOUND_LAYER)
+	if(health < maxHealth * 0.5) //Injuries appear at less than 50% health
+		var/image/I
+		var/is_dead = FALSE
+		if(stat == DEAD)
+			is_dead = TRUE
+		if(resting)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_resting", "layer"=-X_WOUND_LAYER)
+		else if(sleeping || is_dead)
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded_sleeping", "layer"=-X_WOUND_LAYER)
+		else
+			I = image("icon"='icons/Xeno/2x2_wound_overlays.dmi', "icon_state"="ravager_wounded", "layer"=-X_WOUND_LAYER)
+
+		overlays_standing[X_WOUND_LAYER] = I
+		apply_overlay(X_WOUND_LAYER)
+
+
+/mob/living/carbon/Xenomorph/proc/update_wounds()
+	remove_overlay(X_WOUND_LAYER)
```

Comment:
```suggestion
	return FALSE
```
No need to remove any overlay if we don't add it or this is not invoked by any other proc.